src/emulator/auth/operations.ts

```diff
@@ -1232,8 +1231,20 @@ export function setAccountInfoImpl(
     }
   }
 
+  if (reqBody.linkProviderUserInfo) {
+    if (!reqBody.linkProviderUserInfo.providerId?.length) {
+      throw new Error("providerId is required and must be a non-empty string.")
```

Comment:
These should use `assert` to throw a client error instead of internal error, but I don't know what the correct error codes should be. Maybe we can settle with `assert(condition, "TODO_ERROR_CODE : missing providerId")` for now